js/reference.js

```diff
@@ -376,59 +382,92 @@ var ERMrest = (function(module) {
-                var keycols = keys[0].colset.columns;
-
-                // append @sort(..) to uri
-                var sortby;
-                if (this._sort && this._sort.length > 0){
-                    sortby = this._sort;
-                } else if (this._table.annotations.contains(module._annotations.TABLE_DISPLAY) &&
+                // if no sorting provided, use schema defined sort if that's present
+                // If neither one present, use shortestkey
+                if (!this._sort || (this._sort.length === 0)) {
+                    if (this._table.annotations.contains(module._annotations.TABLE_DISPLAY) &&
                         this._table.annotations.get(module._annotations.TABLE_DISPLAY).contains("row_order")) {
-                    sortby = this._table.annotations.get(module._annotations.TABLE_DISPLAY).get("row_order");
+                        this._sort = this._table.annotations.get(module._annotations.TABLE_DISPLAY).get("row_order");
+                    } else {
+                        // use shortest key as sort
+                        this._sort = [];
+                        for (var sk = 0; sk < this._shortestKey.length; sk++) {
+                            var col = this._shortestKey[sk].name;
+                            this._sort.push({"column":col, "descending":false});
+                        }
+                    }
                 }
 
-                if (sortby) {
+                // insert @sort()
+                // get a list of columns in this._sort
+                var sortCols = this._sort.map(function(sort) {
+                    return sort.column;});
+
+                for (var d = 0; d < this._sort.length; d++) {
+                    // column name
+                    var sortCol = this._sort[d].column;
+                    var order = (this._sort[d].descending ? "::desc::" : "");
+                    if (d === 0)
+                        uri = uri + "@sort(" + module._fixedEncodeURIComponent(sortCol) + order;
+                    else
+                        uri = uri + "," + module._fixedEncodeURIComponent(sortCol) + order;
+                }
 
-                    // get a list of columns in sortby
-                    var sortCols = sortby.map(function(sort) {
-                        return sort.column;});
+                // ermrest requires key columns to be in sort param
+                // add them if they are missing
+                for (var i = 0; i < this._shortestKey.length; i++) { // all the key columns
+                    var col = this._shortestKey[i].name;
+                    // add if key col is not in the sortby list
+                    if (!sortCols.includes(col)) {
+                        this._sort.push({"column":module._fixedEncodeURIComponent(col), "descending":false}); // add key to sort
+                        uri = uri + "," + module._fixedEncodeURIComponent(col);
+                    }
+                }
+                uri = uri + ")";
```

Comment:
OK, but, the default or *only* key might be composite, so you can't just check for presence of one column...